common/src/types/v0/store/node.rs

```diff
@@ -60,11 +68,29 @@ impl NodeSpec {
     pub fn set_endpoint(&mut self, endpoint: String) {
         self.endpoint = endpoint
     }
+    /// Cordon node by applying the label.
+    pub fn cordon(&mut self, label: String) {
+        self.cordon_labels.push(label);
+    }
+    /// Uncordon node by removing the corresponding label.
+    pub fn uncordon(&mut self, label: String) {
+        if let Ok(index) = self.cordon_labels.binary_search(&label) {
```

Comment:
Question: Does `binary_search()` method sort the array internally before searching? If not do we need to sort here?

Reply:
I don't believe it sorts it internally. However, I don't think it matters. We disallow multiple instances of the same label being added, so we should only ever have one.

Reply:
Yes you are correct. Let me find an alternative.

Reply:
Good catch @Abhinandan-Purkait. I'm now using `.iter().position()` to get the index so that I can remove the label.